WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/OrderCustomerSection.swift

```diff
@@ -0,0 +1,141 @@
+import SwiftUI
+import Networking
+
+/// Represents the Customer section
+///
+struct OrderCustomerSection: View {
+    let geometry: GeometryProxy
+
+    /// View model to drive the view content
+    @ObservedObject var viewModel: NewOrderViewModel
+
+    @State private var showAddressForm: Bool = false
+
+    var body: some View {
+        Group {
+            Divider()
+
+            VStack(alignment: .leading) {
+                HStack(alignment: .top) {
+                    Text(Localization.customer)
+                        .headlineStyle()
+
+                    Spacer()
+
+                    if viewModel.customerDataViewModel.isDataAvailable {
+                        Button(Localization.editButton) {
+                            showAddressForm.toggle()
+                        }
+                        .buttonStyle(LinkButtonStyle())
+                        .fixedSize(horizontal: true, vertical: true)
+                        .padding(.top, -Layout.linkButtonTopPadding) // remove padding to align button title to the top
+                        .padding(.trailing, -Layout.linkButtonTrailingPadding) // remove padding to align button title to the side
+                    }
+                }
+
+                if !viewModel.customerDataViewModel.isDataAvailable {
+                    createCustomerView
+                } else {
+                    if let fullName = viewModel.customerDataViewModel.fullName {
+                        Text(fullName)
+                            .bodyStyle()
+
+                        if let email = viewModel.customerDataViewModel.email {
+                            Spacer()
+                            Text(email)
+                                .footnoteStyle()
+                        }
```

Comment:
Can't we have en email without a full name?

Reply:
<img width="408" alt="Screen Shot 2021-12-27 at 5 50 12 PM" src="https://user-images.githubusercontent.com/562080/147512093-72c4c2ae-2df8-4759-9da2-1434a86da620.png">

I see that full name is always "valid" even if I don't enter any name, I think we should not render that empty space, don't you think?

Reply:
Added fullName check in 044bd4d3463856d357e19fdafa33c47c5ea992d3.
Fixed layout in 862016873fd5d5b69d7d4b4bfc9a35318dcc535c.